plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/partition/HudiPartitionHiveInfo.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.hudi.partition;
+
+import io.trino.plugin.hive.HiveColumnHandle;
+import io.trino.plugin.hive.HivePartitionKey;
+import io.trino.plugin.hive.metastore.Column;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.plugin.hive.metastore.Partition;
+import io.trino.plugin.hive.metastore.Table;
+import io.trino.plugin.hive.util.HiveUtil;
+import io.trino.plugin.hudi.HudiUtil;
+import io.trino.spi.predicate.TupleDomain;
+import org.apache.hadoop.fs.Path;
+import org.apache.hudi.common.fs.FSUtils;
+import org.apache.hudi.exception.HoodieIOException;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static java.lang.String.format;
+import static java.util.Objects.isNull;
+
+public class HudiPartitionHiveInfo
```

Comment:
This is because of `PartitionValueExtractor` in Hudi. Partition info can be different depending on whether on not users used any partition extractor while syncing Hudi table to Hive. By default, when users don't apply any custom partition extractor, then tables get synced to Hive metastore using hive-style partitioning and partition info is fetched from metastore. This corresponds to `HudiPartitionHiveInfo` class. While in the other case when custom extractor logic needs to be applied, `HudiPartitionInternalInfo` is used.

I can rename as suggested.